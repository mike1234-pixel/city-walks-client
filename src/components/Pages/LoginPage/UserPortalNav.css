/** @format */
.user-portal {
  text-align: left;
}
@media (min-width: 767px) and (max-width: 991px) {
  .user-portal .nav-item i {
    display: none;
  }
  .user-portal .user-portal-link {
    padding: 0;
  }
}

label[for=hamburger2] {
  margin-right: 15px;
}
@media (min-width: 767px) and (max-width: 991px) {
  label[for=hamburger2] {
    display: none;
  }
}

.user-portal-logo {
  font-family: var(--logo-font);
  font-size: 130%;
}
@media (max-width: 767px) {
  .user-portal-logo {
    margin-left: 15px;
  }
}

.login-panel {
  background: black;
}

@media (max-width: 767px) {
  .login-panel {
    line-height: 2.5;
  }
}
.user-portal-link {
  padding: 20px;
  color: var(--blue);
}
.user-portal-link:hover, .user-portal-link:active, .user-portal-link:focus {
  color: white !important;
}

/*# sourceMappingURL=UserPortalNav.css.map */
