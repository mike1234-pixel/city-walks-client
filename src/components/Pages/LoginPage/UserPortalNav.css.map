{"version":3,"sourceRoot":"","sources":["UserPortalNav.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;;AACA;EAEI;IACE;;EAGJ;IACE;;;;AAKN;EACE;;AACA;EAFF;IAGI;;;;AAIJ;EACE;EACA;;AACA;EAHF;IAII;;;;AAIJ;EACE;;;AAGF;EACE;IACE;;;AAIJ;EACE;EACA;;AACA;EAGE","file":"UserPortalNav.css"}