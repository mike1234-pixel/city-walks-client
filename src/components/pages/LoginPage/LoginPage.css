/** @format */

.user-portal-logo {
  font-family: var(--logo-font);
  font-size: 130%;
}

.login-panel {
  background: var(--blue);
}

.user-portal-link {
  padding: 20px;
}
