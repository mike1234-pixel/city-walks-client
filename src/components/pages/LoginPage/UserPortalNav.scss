/** @format */

.user-portal {
  text-align: left;
  @media (min-width: 767px) and (max-width: 991px) {
    .nav-item {
      i {
        display: none;
      }
    }
    .user-portal-link {
      padding: 0;
    }
  }
}

label[for="hamburger2"] {
  margin-right: 15px;
  @media (min-width: 767px) and (max-width: 991px) {
    display: none;
  }
}

.user-portal-logo {
  font-family: var(--logo-font);
  font-size: 130%;
  @media (max-width: 767px) {
    margin-left: 15px;
  }
}

.login-panel {
  background: black;
}

@media (max-width: 767px) {
  .login-panel {
    line-height: 2.5;
  }
}

.user-portal-link {
  padding: 20px;
  color: var(--blue);
  &:hover,
  &:active,
  &:focus {
    color: white !important;
  }
}
