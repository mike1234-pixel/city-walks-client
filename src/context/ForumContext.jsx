import { createContext, useState } from "react"
import BoardBox from "../components/pages/LoginPage/LoggedInView/Components/BoardBox/BoardBox"

export const ForumContext = createContext();

export const ForumContextProvider = (props) => {
 
    const [loadingBoards, setLoadingBoards] = useState(true)
    const [boards, setBoards] = useState([])
    const [selectedThreads, setSelectedThreads] = useState([])

    const [currentBoardName, setCurrentBoardName] = useState("")
    const [currentBoardId, setCurrentBoardId] = useState("")

    let displayBoards;

    if (!loadingBoards) {
    displayBoards = boards.map((board, index) => {
        return (
            <BoardBox boardId={board._id} name={board.name} description={board.description} index={index} key={index} />
                )
        })
        // create routes for boards with Thread as the view component
    }

    return (
        <ForumContext.Provider 
            value={{
                setBoards,
                boards, // data
                setLoadingBoards,
                loadingBoards,
                displayBoards,
                setSelectedThreads,
                selectedThreads, 
                setCurrentBoardName,
                currentBoardName,
                setCurrentBoardId,
                currentBoardId,
            }}>
            {props.children}
        </ForumContext.Provider>
    )
}

