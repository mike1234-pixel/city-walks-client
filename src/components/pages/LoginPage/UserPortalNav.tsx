import React, { useState, useEffect } from "react"
import {
  MDBIcon,
  MDBNavbar,
  MDBNavbarBrand,
  MDBNavbarNav,
  MDBNavItem,
  MDBHamburgerToggler,
  MDBCollapse,
  MDBContainer
} from "mdbreact";
import { Link, useHistory } from "react-router-dom"
import { connect } from 'react-redux'
import { Dispatch } from "redux";
import RootState from "../../../types/State/Root/State"
import "./UserPortalNav.scss"

interface Props {
  loggedIn: boolean;
  setLoggedIn: Function;
  setUserId: Function;
  setUserFirstName: Function;
  setUserLastName: Function;
}

const UserPortalNav: React.FC<Props> = (props: Props) => {

  const { loggedIn, setLoggedIn, setUserId, setUserFirstName, setUserLastName } = props

  const [toggleLoginPanel, setToggleLoginPanel] = useState<boolean>(false);

  const handleToggleLoginPanel: () => void = () => {
    setToggleLoginPanel(!toggleLoginPanel);
  };

  const pushSlug: Function = useHistory().push

  const logOut: () => void = () => {
    localStorage.clear()
    localStorage.setItem('popupVisible', 'false')
    setLoggedIn(false)
    setUserId('')
    setUserFirstName('')
    setUserLastName('')
    alert("Logged out successfully.")
    pushSlug("/forum");
  }

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  return (
    <div className="user-portal" key="login-page">
      {loggedIn ? (
        <div>
          <MDBNavbar className="login-panel" dark expand="md">
            <MDBContainer>
              <MDBNavbarBrand>
                <strong className="white-text user-portal-logo">User Portal</strong>
              </MDBNavbarBrand>
              <MDBHamburgerToggler
                color="#fff"
                className="hamburger1"
                id="hamburger2"
                onClick={handleToggleLoginPanel}
              />
              <MDBCollapse
                id="navbarCollapse3"
                isOpen={toggleLoginPanel}
                navbar
              >
                <MDBNavbarNav right>
                  <MDBNavItem>
                    <a className="user-portal-link" onClick={logOut}>
                      Logout <MDBIcon icon="key" />
                    </a>
                  </MDBNavItem>
                </MDBNavbarNav>
              </MDBCollapse>
            </MDBContainer>
          </MDBNavbar>
        </div>
      ) : (
        <div>

          <MDBNavbar className="login-panel" dark expand="md">
            <MDBContainer>
              <MDBNavbarBrand>
                <strong className="white-text user-portal-logo">User Portal</strong>
              </MDBNavbarBrand>
              <MDBHamburgerToggler
                color="#fff"
                className="hamburger1"
                id="hamburger2"
                onClick={handleToggleLoginPanel}
              />
              <MDBCollapse
                id="navbarCollapse3"
                isOpen={toggleLoginPanel}
                navbar
              >
                <MDBNavbarNav justify="true">
                  <MDBNavItem>
                    <Link to="/forum/login" className="user-portal-link">Login <MDBIcon icon="key" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                    <Link to="/forum/register" className="user-portal-link">Register <MDBIcon far icon="edit" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                    <Link to="/forum/verify" className="user-portal-link">Resend Email <MDBIcon far icon="envelope" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                    <Link to="/forum/reset-password" className="user-portal-link">Reset Password <MDBIcon icon="unlock-alt" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                    <Link to="/forum/forgot-password" className="user-portal-link">Forgot Password <MDBIcon far icon="question-circle" /></Link>
                  </MDBNavItem>
                </MDBNavbarNav>
              </MDBCollapse>
            </MDBContainer>
          </MDBNavbar>

        </div>

      )}
      <br />
    </div>
  );
};

const mapStateToProps: (state: RootState) => void = (state) => ({
  loggedIn: state.loginState.loggedIn,
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    setLoggedIn: (boolValue: boolean) => dispatch({ type: 'SET_LOGGED_IN', boolValue }),
    setUserId: (userID: string) => dispatch({ type: 'SET_USER_ID', userID }),
    setUserFirstName: (userFirstName: string) => dispatch({ type: 'SET_USER_FIRST_NAME', userFirstName }),
    setUserLastName: (userLastName: string) => dispatch({ type: 'SET_USER_LAST_NAME', userLastName }),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(UserPortalNav);
