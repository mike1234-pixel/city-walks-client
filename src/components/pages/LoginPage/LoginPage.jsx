/** @format */

import { useState, useEffect, useContext } from "react";
import {
  MDBIcon,
  MDBNavbar,
  MDBNavbarBrand,
  MDBNavbarNav,
  MDBNavItem,
  MDBHamburgerToggler,
  MDBCollapse,
} from "mdbreact";
import { LoginContext } from "../../../context/LoginContext";
import "./LoginPage.css";

const LoginPage = () => {
  const [toggleLoginPanel, setToggleLoginPanel] = useState(false);

  const handleToggleLoginPanel = () => {
    setToggleLoginPanel(!toggleLoginPanel);
  };

  const { logOut, displayForm, setForm, loggedIn } = useContext(LoginContext);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  return (
    <div className="user-portal min-page-height" key="login-page">
        {loggedIn ? (
          <div>
          <MDBNavbar className="login-panel" dark expand="md">
              <MDBNavbarBrand>
                <strong className="white-text user-portal-logo">User Portal</strong>
              </MDBNavbarBrand>
              <MDBHamburgerToggler
                color="#fff"
                className="hamburger1"
                id="hamburger2"
                onClick={handleToggleLoginPanel}
              />
              <MDBCollapse
                id="navbarCollapse3"
                isOpen={toggleLoginPanel}
                navbar
              >
                <MDBNavbarNav right>
                  <MDBNavItem>
                    <a className="user-portal-link" onClick={logOut}>
                      Logout <MDBIcon icon="key" />
                    </a>
                  </MDBNavItem>
                </MDBNavbarNav>
              </MDBCollapse>
            </MDBNavbar>
          </div>
        ) : (
          <div>
            <MDBNavbar className="login-panel" dark expand="md">
              <MDBNavbarBrand>
                <strong className="white-text user-portal-logo">User Portal</strong>
              </MDBNavbarBrand>
              <MDBHamburgerToggler
                color="#fff"
                className="hamburger1"
                id="hamburger2"
                onClick={handleToggleLoginPanel}
              />
              <MDBCollapse
                id="navbarCollapse3"
                isOpen={toggleLoginPanel}
                navbar
              >
                <MDBNavbarNav justify="true">
                  <MDBNavItem>
                    <a className="user-portal-link" onClick={() => setForm("loginForm")}>
                      Login <MDBIcon icon="key" />
                    </a>
                  </MDBNavItem>
                  <MDBNavItem>
                    <a className="user-portal-link"
                      onClick={() => setForm("registrationForm")}
                    >
                      Register <MDBIcon far icon="edit" />
                    </a>
                  </MDBNavItem>
                  <MDBNavItem>
                    <a className="user-portal-link"
                      onClick={() => setForm("verificationForm")}
                    >
                      Resend Account Verification Email <MDBIcon far icon="envelope" />
                    </a>
                  </MDBNavItem>
                  <MDBNavItem>
                    <a className="user-portal-link"
                      onClick={() => setForm("resetPasswordForm")}
                    >
                      Reset Password <MDBIcon icon="unlock-alt" />
                    </a>
                  </MDBNavItem>
                  <MDBNavItem>
                    <a className="user-portal-link"
                      onClick={() => setForm("forgotPasswordForm")}
                    >
                      Forgot Password <MDBIcon far icon="question-circle" />
                    </a>
                  </MDBNavItem>
                </MDBNavbarNav>
              </MDBCollapse>
            </MDBNavbar>
          </div>
        )}
      <br />
      {displayForm}
    </div>
  );
};

export default LoginPage;
