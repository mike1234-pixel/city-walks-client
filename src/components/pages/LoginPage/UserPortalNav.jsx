/** @format */

import { useState, useEffect, useContext } from "react";
import {
  MDBIcon,
  MDBNavbar,
  MDBNavbarBrand,
  MDBNavbarNav,
  MDBNavItem,
  MDBHamburgerToggler,
  MDBCollapse,
} from "mdbreact";
import { Link } from "react-router-dom"
import { LoginContext } from "../../../context/LoginContext";
import "./UserPortalNav.css";

const UserPortalNav = () => {
  const [toggleLoginPanel, setToggleLoginPanel] = useState(false);

  const handleToggleLoginPanel = () => {
    setToggleLoginPanel(!toggleLoginPanel);
  };

  const { logOut, loggedIn } = useContext(LoginContext);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  return (
    <div className="user-portal" key="login-page">
        {loggedIn ? (
          <div>
          <MDBNavbar className="login-panel" dark expand="md">
              <MDBNavbarBrand>
                <strong className="white-text user-portal-logo">User Portal</strong>
              </MDBNavbarBrand>
              <MDBHamburgerToggler
                color="#fff"
                className="hamburger1"
                id="hamburger2"
                onClick={handleToggleLoginPanel}
              />
              <MDBCollapse
                id="navbarCollapse3"
                isOpen={toggleLoginPanel}
                navbar
              >
                <MDBNavbarNav right>
                  <MDBNavItem>
                    <a className="user-portal-link" onClick={logOut}>
                      Logout <MDBIcon icon="key" />
                    </a>
                  </MDBNavItem>
                </MDBNavbarNav>
              </MDBCollapse>
            </MDBNavbar>
          </div>
        ) : (
          <div>
            <MDBNavbar className="login-panel" dark expand="md">
              <MDBNavbarBrand>
                <strong className="white-text user-portal-logo">User Portal</strong>
              </MDBNavbarBrand>
              <MDBHamburgerToggler
                color="#fff"
                className="hamburger1"
                id="hamburger2"
                onClick={handleToggleLoginPanel}
              />
              <MDBCollapse
                id="navbarCollapse3"
                isOpen={toggleLoginPanel}
                navbar
              >
                <MDBNavbarNav justify="true">
                  <MDBNavItem>
                  <Link to="/boards/login" className="user-portal-link">Login <MDBIcon icon="key" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                  <Link to="/boards/register" className="user-portal-link">Register <MDBIcon far icon="edit" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                  <Link to="/boards/verify" className="user-portal-link">Resend Email <MDBIcon far icon="envelope" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                  <Link to="/boards/reset-password" className="user-portal-link">Reset Password <MDBIcon icon="unlock-alt" /></Link>
                  </MDBNavItem>
                  <MDBNavItem>
                  <Link to="/boards/forgot-password" className="user-portal-link">Forgot Password <MDBIcon far icon="question-circle" /></Link>
                  </MDBNavItem>
                </MDBNavbarNav>
              </MDBCollapse>
            </MDBNavbar>
          </div>
        )}
      <br />
    </div>
  );
};

export default UserPortalNav;
