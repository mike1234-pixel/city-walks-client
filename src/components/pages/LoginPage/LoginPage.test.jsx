import renderer from 'react-test-renderer'
import { LoginContextProvider } from "../../../context/LoginContext";
import { ForumContextProvider } from "../../../context/ForumContext"
import LoginPage from "./LoginPage"

// snapshot test
it('LoginPage Page renders correctly', () => {  

  const tree = renderer
    .create(
        <LoginContextProvider>
            <ForumContextProvider>
                <LoginPage />
            </ForumContextProvider>
        </LoginContextProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});