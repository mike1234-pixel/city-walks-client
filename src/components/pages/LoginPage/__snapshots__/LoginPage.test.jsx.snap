// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Page renders correctly 1`] = `
<div
  className="user-portal min-page-height"
>
  <div>
    <nav
      className="navbar-dark navbar navbar-expand-md login-panel"
      data-test="navbar"
      role="navigation"
    >
      <div
        className="navbar-brand"
        data-test="navbar-brand"
      >
        <strong
          className="white-text user-portal-logo"
        >
          User Portal
        </strong>
      </div>
      <input
        className="hamburger-button__checkbox"
        data-test="hamburger-toggler"
        defaultChecked={false}
        id="hamburger2"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="hamburger-button__button hamburger1"
        htmlFor="hamburger2"
        id="nav-icon1"
      >
        <span
          style={
            Object {
              "background": "#fff",
            }
          }
        />
        <span
          style={
            Object {
              "background": "#fff",
            }
          }
        />
        <span
          style={
            Object {
              "background": "#fff",
            }
          }
        />
      </label>
      <div
        className="collapse navbar-collapse"
        data-test="collapse"
        id="navbarCollapse3"
        style={Object {}}
      >
        <ul
          className="navbar-nav justify-content-around w-100"
          data-test="navbar-nav"
          justify="true"
        >
          <li
            className="nav-item"
            data-test="nav-item"
          >
            <a
              className="user-portal-link"
              onClick={[Function]}
            >
              Login 
              <i
                className="fa fa-key"
                data-test="fa"
              />
            </a>
          </li>
          <li
            className="nav-item"
            data-test="nav-item"
          >
            <a
              className="user-portal-link"
              onClick={[Function]}
            >
              Register 
              <i
                className="far fa-edit"
                data-test="fa"
              />
            </a>
          </li>
          <li
            className="nav-item"
            data-test="nav-item"
          >
            <a
              className="user-portal-link"
              onClick={[Function]}
            >
              Resend Account Verification Email 
              <i
                className="far fa-envelope"
                data-test="fa"
              />
            </a>
          </li>
          <li
            className="nav-item"
            data-test="nav-item"
          >
            <a
              className="user-portal-link"
              onClick={[Function]}
            >
              Reset Password 
              <i
                className="fa fa-unlock-alt"
                data-test="fa"
              />
            </a>
          </li>
          <li
            className="nav-item"
            data-test="nav-item"
          >
            <a
              className="user-portal-link"
              onClick={[Function]}
            >
              Forgot Password 
              <i
                className="far fa-question-circle"
                data-test="fa"
              />
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <br />
  <div>
    <div
      className="page-heading-container"
    >
      <h1
        className="page-heading"
      >
        Forum
      </h1>
      <h2
        className="login-heading"
      >
        Welcome back 
        
      </h2>
      <p>
        Here you can join in the discussion and put forward your own recomendations. Discussions are subdivided by category:
      </p>
    </div>
    <p>
      Loading...
    </p>
  </div>
</div>
`;
